src/lib/storage/storage_manager.cpp

```diff
@@ -638,8 +667,82 @@ PersistedSegmentEncodingType StorageManager::resolve_persisted_segment_encoding_
   return persisted_vector_type_id;
 }
 
-void StorageManager::export_compressed_vector(const CompressedVectorType type, const BaseCompressedVector& compressed_vector,
-                              std::ofstream& ofstream) {
+void StorageManager::_serialize_table_files_mapping() {
+  for (const auto& mapping : _tables_current_persistence_file_mapping) {
+    const auto table = get_table(mapping.first);
+    const auto column_count = table->column_count();
+    const auto chunk_count = mapping.second.file_index * MAX_CHUNK_COUNT_PER_FILE + mapping.second.current_chunk_count;
+    json table_json = {{"file_count", mapping.second.file_index + 1},
+                       {"chunk_count", chunk_count},
+                       {"column_count", static_cast<uint32_t>(table->column_count())}};
+
+    const auto column_definitions = table->column_definitions();
+    auto columns_json = json::array();
+    for (auto index = size_t{0}; index < column_count; ++index) {
+      const json column_object = {
+          {"column_name", column_definitions[index].name},
+          {"data_type", magic_enum::enum_name(column_definitions[index].data_type)},
+          {"nullable", column_definitions[index].nullable},
+      };
+      columns_json.push_back(column_object);
+    }
+    table_json["columns"] = columns_json;
+
+    _storage_json[mapping.first] = table_json;
+  }
+}
+
+void StorageManager::save_storage_json_to_disk() {
+  _serialize_table_files_mapping();
+  std::ofstream output_file(_cache_directory + _storage_json_name, std::ios::trunc);
+  const auto json_serialized = _storage_json.dump(4);
+  output_file << json_serialized;
+  output_file.close();
+}
+
+std::vector<TableColumnDefinition> StorageManager::get_table_column_definitions_from_json(
```

Comment:
private?

Reply:
Cannot be private because this function is called within the abstractTableGenerator to recreate the tables.


```
465 // Create column definitions
466   const auto column_definitions = storage_manager.get_table_column_definitions_from_json(table_name);
```